Description: Adapt to PHPUnit 8.x and 9.x API.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/ActiveSyncTests.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/ActiveSyncTests.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/ActiveSyncTests.php
@@ -22,7 +22,7 @@ class Horde_Core_ActiveSyncTests extends
     protected $_mailboxes;
     protected $_special;
 
-    public function setUp()
+    public function setUp(): void
     {
         $this->_auth = $this->getMockSkipConstructor('Horde_Auth_Auto');
         $this->_state = $this->getMockSkipConstructor('Horde_ActiveSync_State_Sql');
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/Factory/GroupTest.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/Factory/GroupTest.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/Factory/GroupTest.php
@@ -23,7 +23,7 @@
  * @author   Gunnar Wrobel <wrobel@pardus.de>
  * @license  http://www.horde.org/licenses/lgpl21 LGPL 2.1
  */
-class Horde_Core_Factory_GroupTest extends PHPUnit_Framework_TestCase
+class Horde_Core_Factory_GroupTest extends Horde_Test_Case
 {
     public function testMock()
     {
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/Factory/KolabServerTest.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/Factory/KolabServerTest.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/Factory/KolabServerTest.php
@@ -23,9 +23,9 @@
  * @author   Gunnar Wrobel <wrobel@pardus.de>
  * @license  http://www.horde.org/licenses/lgpl21 LGPL 2.1
  */
-class Horde_Core_Factory_KolabServerTest extends PHPUnit_Framework_TestCase
+class Horde_Core_Factory_KolabServerTest extends Horde_Test_Case
 {
-    public function setUp()
+    public function setUp(): void
     {
         $this->markTestIncomplete('Needs some love');
         $GLOBALS['conf']['kolab']['server']['basedn'] = 'test';
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/Factory/KolabSessionTest.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/Factory/KolabSessionTest.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/Factory/KolabSessionTest.php
@@ -23,9 +23,9 @@
  * @author   Gunnar Wrobel <wrobel@pardus.de>
  * @license  http://www.horde.org/licenses/lgpl21 LGPL 2.1
  */
-class Horde_Core_Factory_KolabSessionTest extends PHPUnit_Framework_TestCase
+class Horde_Core_Factory_KolabSessionTest extends Horde_Test_Case
 {
-    public function setUp()
+    public function setUp(): void
     {
         $this->markTestIncomplete('Needs some love');
     }
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/NlsconfigTest.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/NlsconfigTest.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/NlsconfigTest.php
@@ -20,7 +20,7 @@
  */
 class Horde_Core_NlsconfigTest extends Horde_Test_Case
 {
-    public function setUp()
+    public function setUp(): void
     {
         $GLOBALS['session'] = new Horde_Session();
         $GLOBALS['session']->sessionHandler = new Horde_Support_Stub();
@@ -58,6 +58,8 @@ class Horde_Core_NlsconfigTest extends H
      */
     public function testGet($key, $expected)
     {
+        $this->expectNotToPerformAssertions();
+
         $nls = new Horde_Registry_Nlsconfig();
     }
 
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/RegistryTest.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/RegistryTest.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/RegistryTest.php
@@ -18,11 +18,11 @@
  * @license  http://www.horde.org/licenses/lgpl21 LGPL 2.1
  * @package  Core
  */
-class Horde_Core_RegistryTest extends PHPUnit_Framework_TestCase
+class Horde_Core_RegistryTest extends Horde_Test_Case
 {
     protected $_tmpdir;
 
-    public function tearDown()
+    public function tearDown(): void
     {
         if (is_dir($this->_tmpdir)) {
             rmdir($this->_tmpdir);
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/UrlTest.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/UrlTest.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/UrlTest.php
@@ -6,7 +6,7 @@
  * @package    Core
  * @subpackage UnitTests
  */
-class Horde_Core_UrlTest extends PHPUnit_Framework_TestCase
+class Horde_Core_UrlTest extends Horde_Test_Case
 {
     public function testUrl()
     {
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/phpunit.xml
===================================================================
--- /dev/null
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/phpunit.xml
@@ -0,0 +1 @@
+<phpunit bootstrap="bootstrap.php"></phpunit>
Index: php-horde-core/Horde_Core-2.31.18/test/Horde/Core/SmartmobileUrlTest.php
===================================================================
--- php-horde-core.orig/Horde_Core-2.31.18/test/Horde/Core/SmartmobileUrlTest.php
+++ php-horde-core/Horde_Core-2.31.18/test/Horde/Core/SmartmobileUrlTest.php
@@ -8,11 +8,9 @@
  */
 class Horde_Core_SmartmobileUrlTest extends Horde_Test_Case
 {
-    /**
-     * @expectedException InvalidArgumentException
-     */
     public function testInvalidParamter()
     {
+        $this->expectException('InvalidArgumentException');
         new Horde_Core_Smartmobile_Url('test');
     }
 
