Description: Fix PHP 8.2 deprecation warnings.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

--- a/Horde_Core-2.31.18/test/Horde/Core/NlsconfigTest.php
+++ b/Horde_Core-2.31.18/test/Horde/Core/NlsconfigTest.php
@@ -18,6 +18,7 @@
  * @license  http://www.horde.org/licenses/lgpl21 LGPL 2.1
  * @package  Core
  */
+#[\AllowDynamicProperties]
 class Horde_Core_NlsconfigTest extends Horde_Test_Case
 {
     public function setUp(): void
